packages/mui-material/src/styles/useThemeProps.js

```diff
@@ -1,6 +1,7 @@
 import { useThemeProps as systemUseThemeProps } from '@mui/system';
 import defaultTheme from './defaultTheme';
+import IDENTIFIER from './identifier';
 
 export default function useThemeProps({ props, name }) {
-  return systemUseThemeProps({ props, name, defaultTheme });
+  return systemUseThemeProps({ props, name, defaultTheme, identifier: IDENTIFIER });
```

Comment:
This is still not 100% clear to me.

Reply:
> Is it required to use the identifier by default?

It is required. Otherwise, the built-in components could get the wrong theme.

If we don't do this, how do these APIs know when to get the right theme?

Providing as a argument will be much more confusing because you have to specify it every time and it breaks existing project if they turn on theme scoping, e.g.:

```js
// this works fine
<ThemeProvider theme={theme}>
  {…some component call useTheme() from @mui/material/styles}
</ThemeProvider>


// this will break
<ThemeProvider theme={{ [THEME_ID]: theme }}>
  {…some component call useTheme() from @mui/material/styles}
</ThemeProvider>
// you have to fix to `useTheme(THEME_ID)`
```